### PR TITLE
Auto detect user language #177

### PR DESCRIPTION
**Pull request by stefan-kolb**
JabRef now uses the system locale as default with this PR.

I simplified some methods in the process and implemented a fallback to "en" when there is no bundle for the default locale.

Any suggestions?

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/l10n/Localization.java [hunk 1]
```diff
@@ -17,28 +19,37 @@
...
+            }
+        } catch(MissingResourceException e) {
+            LOGGER.warn("Fallback to system locale <" + defaultLocale + "> failed, using locale <en> instead");
+
+            locale = new Locale("en");
+            messages = ResourceBundle.getBundle(RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
+            menuTitles = ResourceBundle.getBundle(MENU_RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
+            intMessages = ResourceBundle.getBundle(INTEGRITY_RESOURCE_PREFIX, locale, new EncodingControl("UTF-8"));
+        } finally {
+            Locale.setDefault(locale);
```
Comment (stefan-kolb) [inline src/main/java/net/sf/jabref/logic/l10n/Localization.java]: Are these two lines really needed? What do they change exactly? maybe we should add a clarifying comment!

**Comment by matthiasgeiger** [comment]
:+1: 
Should we clarify [in the wiki](https://github.com/JabRef/jabref/wiki/Translating-JabRef) that a new translation requires all three properties files (even if no keys are translated)? 

@mlep This might be good to know for you, too.

**Comment by mlep** [comment]
Noted!
I can be the one adding it to the wiki if you want.

**Comment by matthiasgeiger** [comment]
Okay, thank you!

**Comment by mlep** [comment]
Reading the wiki, it seems clear to me that the "Step-by-Step guide" (using Popeye) states that the 3 files must be created.

**Pull request merged**